src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -125,38 +125,31 @@ class IterationChecker final {
             const auto rateLimiterName =
                 phaseContext["RateLimiterName"].maybe<std::string>().value_or(defaultRLName.str());
 
-            if (!_doesBlock) {
-                throw InvalidConfigurationException(
-                    "GlobalRate must be specified alongside either Duration or Repeat, otherwise "
-                    "there's no guarantee the rate limited operation will run in the correct "
-                    "phase");
-            }
             _rateLimiter =
                 phaseContext.workload().getRateLimiter(rateLimiterName, rateSpec.value());
         }
     }
 
     constexpr void limitRate(const SteadyClock::time_point referenceStartingPoint,
                              const int64_t currentIteration,
+                             Orchestrator& orchestrator,
                              const PhaseNumber inPhase) {
-        // This function is called after each iteration, so we never rate limit the
-        // first iteration. This means the number of completed operations is always
-        // `n * GlobalRateLimiter::_burstSize + m` instead of an exact multiple of
-        // _burstSize. `m` here is the number of threads using the rate limiter.
```

Comment:
Removed this comment because it's wrong, TIG-1825 moved to rate limiter to the == operator that runs before each iteration.